CalibTracker/SiStripHitEfficiency/plugins/SiStripHitEfficiencyWorker.cc

```diff
@@ -845,23 +892,42 @@ void SiStripHitEfficiencyWorker::fillForTraj(const TrajectoryAtInvalidHit& tm,
 
         // hot/cold maps of hits that are expected but not found
         if (badflag) {
-          if (layer > 0 && layer <= 10) {
-            // 1-4: TIB, 4-10: TOB
-            h_hotcold[layer - 1]->Fill(360. - ::calcPhi(tm.globalX(), tm.globalY()), tm.globalZ(), 1.);
-          } else if (layer > 10 && layer <= 13) {
-            // 11-13: TID, above: TEC
-            const int side = layer > 13 ? (iidd >> 13) & 0x3 : (iidd >> 18) & 0x3;
-            h_hotcold[2 * layer - 13 + side]->Fill(-tm.globalY(), tm.globalX(), 1.);
+          if (layer > bounds::k_LayersStart && layer <= bounds::k_LayersAtTIBEnd) {
+            //We are in the TIB
+            float phi = ::calcPhi(tm.globalX(), tm.globalY());
+            h_hotcold[layer - 1]->Fill(360. - phi, tm.globalZ(), 1.);
+          } else if (layer > bounds::k_LayersAtTIBEnd && layer <= bounds::k_LayersAtTOBEnd) {
+            //We are in the TOB
+            float phi = ::calcPhi(tm.globalX(), tm.globalY());
+            h_hotcold[layer - 1]->Fill(360. - phi, tm.globalZ(), 1.);
+          } else if (layer > bounds::k_LayersAtTOBEnd && layer <= bounds::k_LayersAtTIDEnd) {
+            //We are in the TID
+            //There are 2 different maps here
+            int side = tTopo->tidSide(iidd);  //((iidd >> 13) & 0x3);
+            if (side == 1)
+              h_hotcold[(layer - 1) + (layer - 11)]->Fill(-tm.globalY(), tm.globalX(), 1.);
+            else if (side == 2)
+              h_hotcold[(layer - 1) + (layer - 10)]->Fill(-tm.globalY(), tm.globalX(), 1.);
+          } else if (layer > bounds::k_LayersAtTIDEnd) {
+            //We are in the TEC
+            //There are 2 different maps here
+            int side = tTopo->tecSide(iidd);  //((iidd >> 18) & 0x3);
+            if (side == 1)
```

Comment:
same comment about the comment cleaning (only if you touch the `k_LayersAtTECEnd` part)